execution_engine/src/core/runtime/auction_internal.rs

```diff
@@ -148,6 +151,13 @@ where
         self.context.get_caller()
     }
 
+    fn get_immediate_caller(&self) -> Option<&CallStackElement> {
+        let call_stack = self.call_stack();
+        let mut call_stack_iter = call_stack.iter().rev();
+        call_stack_iter.next();
+        call_stack_iter.next()
```

Comment:
Can you add a comment why you need to call `call_stack_iter.next()` twice before returning?